Smidtj/users

smidtj opened the pull request:
Hey guys! Did a couple things here:
- queryDB is no longer static to conform with the rest of our app
- the way users are managed has changed. When a user is created an actual psql user is created instead of just inserting into an unprotected table.  This user is given a default database called "[username]_default".  
- query.py can now take username and password arguments.  If these are not provided the query will be run as vagrant.  This will allow us to control things like ownership, access, etc.

smidtj committed:
Converted queryDB class to non-static (dynamic?) methods to conform with code standard

smidtj committed:
started users implimentation

smidtj committed:
finished psql users (i.e. creating a user creates an actual psql user)

Commit:
add table display functionality and implimented in the query section of our app

smidtj commented:
Hey guys - threw up another commit to this PR with a few more features:
- query.py now returns a jagged list with the first element being a list of each column header (i.e "id", "city", "population", etc.) and the second element  is a list of lists containing each element of each row. Visually:
  data[0]:
  ("id", "city", "population")
  data[1]:
  [(1, "New York", 8000000), (2, "Los Angeles", 10000000), etc]
- Added tableDisplay.py which can display a table in the format previously described.  Size should not be an issue as whatever fits on the screen is displayed and arrow keys are used to "navigate" around the table.
- Implemented tableDisplay in queryDB.py

smidtj committed:
quick bug fix

CGS-123 commented:
:+1:  Looks great man! Good job.

CGS-123 commented:
I think we should get this in before I start doing any of my work. Quite a bit of work here and I don't want too messy of a rebase.

cornbreadz commented on code/loginscreen.py:
```diff
@@ -49,19 +50,27 @@ def create_user(self):
+        to_query = "SELECT * FROM pg_catalog.pg_roles WHERE rolname LIKE \'%s\'" % self.username
+        rows = query.query(to_query, 'postgres', self.screen)
         
         if rows:
```
This seems to always pass when trying to create a username.  I printed out the contents of rows and it is ['rolname', 'rolsuper', 'rolinherit', 'rolcreaterole', 'rolcreatedb', 'rolcatupdate', 'rolcanlogin', 'rolreplication', 'rolconnlimit', 'rolpassword', 'rolvaliduntil', 'rolconfig', 'oid'][]

Have not been able to create a username because it always hits this if block and states username already exists.

cornbreadz commented:
I can login as root or vagrant but It no longer lists worldb or any of the test databases.  It looks like the user settings are causing the db queries to return slightly different tables like datname  and ['postgres']

smidtj committed:
bug fixes and database selection for queryDB

smidtj commented:
Ok, so we should be good - all the bugs were caused by the changes I made to the query class so I went through the code and added the [1]s to the return values where necessary. Additionally, while I was poking around I added database selection to the queryDB class (before it just ran all queries in the worlddb database).  Should be good to ship now!

cornbreadz commented:
Awesome looks great.  I'll start work on integrating the table display to the list tables menu section Going to go ahead and merge this everything looks great!
